File: frontend/src/types/assignmentGroupTopic.ts
import type { AssignmentGroupApprovalRequestResponse, AssignmentGroupResponse } from './assignmentGroup';

// ============ ENTITIES ============
export interface AssignmentGroupTopic {
  id: number;
  title: string;
  maxGroupsPerTopic: number;
  description?: string | null;
  maxMembers: number;
  minMembers: number;
  createdAt: Date;
  updatedAt: Date;
  assignmentGroups: AssignmentGroupResponse[];
  approvalRequests?: AssignmentGroupApprovalRequestResponse[] | null;
}

// ============ REQUEST TYPES ============
export interface CreateAssignmentGroupTopicRequest {
  title: string;
  maxGroupsPerTopic: number;
  description?: string | null;
  maxMembers: number;
  minMembers: number;
}

export interface UpdateAssignmentGroupTopicRequest {
  title?: string | null;
  maxGroupsPerTopic?: number | null;
  description?: string | null;
  maxMembers?: number | null;
  minMembers?: number | null;
}

// ============ RESPONSE TYPES ============
export interface AssignmentGroupTopicResponse {
  id: number;
  title: string;
  maxGroupsPerTopic: number;
  description?: string | null;
  maxMembers: number;
  minMembers: number;
  assignmentGroups?: AssignmentGroupResponse[] | null;
  approvalRequests?: AssignmentGroupApprovalRequestResponse[] | null;
  createdAt: Date;
  updatedAt: Date;
}

File: backend/Models/Entities/AssignmentGroupTopic.cs
namespace OnlineClassroomManagement.Models.Entities
{
    public class AssignmentGroupTopic
    {
        public int Id { get; set; }
        public string Title { get; set; } = string.Empty;
        public int MaxGroupsPerTopic { get; set; }
        public string? Description { get; set; }
        public int MaxMembers { get; set; } 
        public int MinMembers { get; set; } 
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }
        public List<AssignmentGroup> AssignmentGroups { get; set; } = new();
        public List<AssignmentGroupApprovalRequest> ApprovalRequests { get; set; } = new();
    }
}